### ISSUE TITLE
Whitelist plugins (Urgent)

### ISSUE THREAD

**Issue by Chadori** [Scirra/Construct-3-bugs#5559]
Hi,

Sorry about this but would you please whitelist the plugin below [again](https://github.com/Scirra/Construct-3-bugs/issues/5538), from `0.1.0` to `0.1.1`.

```JS
chadori-mobile-firebase-analytics@0.1.1
>>> [Forked] from "https://www.npmjs.com/package/cordova-plugin-firebase-analytics"
```

The only change I've made is to remove the tag below, hopefully this lessens the work in whitelisting.
```JS
  <config-file target="config.xml" parent="/*">
      <preference name="AndroidXEnabled" value="true" />
      <preference name="GradlePluginGoogleServicesEnabled" value="true" />
      <preference name="GradlePluginGoogleServicesVersion" value="4.3.8" />
  </config-file>
```
I forgot that I always remove this tag every time since it causes conflicts with Construct 3 Build Service's own declarations.
I've fixed the mentioned build failure from #5538, but introduced an old build failure regression.

Again, sorry about this. Unfortunately, this is only reproducible with the Construct 3 Build Service so there's no way to find out if the tag is causing issues since it doesn't affect the Cordova CLI.

### Whitelisting
(As per [whitelisting guideline requirements](https://www.construct.net/en/forum/construct-3/plugin-sdk-10/build-server-cordova-plugin-149266) issued by the Construct Team. To reduce work in whitelisting, if you use one of those plugins please use mine instead, I **always** make sure they are backwards compatible unless otherwise specified.)

I have followed the required guidelines to my knowledge, please let me know if there are issues with my request.
Thank you!

**Comment by AshleyScirra** [comment]
AndroidX is no longer necessary with the latest version of Cordova. Also `GradlePluginGoogleServicesEnabled` can be configured in your plugin via `SetGooglePlayServicesEnabled` in [IPluginInfo](https://www.construct.net/en/make-games/manuals/addon-sdk/reference/iplugininfo). So perhaps you don't need those tags in your plugin?

**Comment by AshleyScirra** [comment]
Actually I'm confused if you mean you're adding or removing them. Either way, I don't see why they would cause a problem with Construct's build service specifically? It just runs the Cordova CLI like any other build.

**Comment by Chadori** [comment]
> AndroidX is no longer necessary with the latest version of Cordova. Also GradlePluginGoogleServicesEnabled can be configured in your Construct plugin via SetGooglePlayServicesEnabled in [IPluginInfo](https://www.construct.net/en/make-games/manuals/addon-sdk/reference/iplugininfo). So perhaps you don't need those tags in your plugin?

Hi @AshleyScirra, I actually removed the tag that I listed, sorry I forgot to be specific about that. I've now updated the report.

> Actually I'm confused if you mean you're adding or removing them. Either way, I don't see why they would cause a problem with Construct's build service specifically? It just runs the Cordova CLI like any other build. And I want to properly understand the problem and make sure we resolve any Construct-specific compatibility issues to prevent them happening in future.

Yeah, I meant removed, I've edited above to be specific.

Yeah, it's actually an interesting thing because it seems contradictory for the Cordova CLI to work but not the C3 Build Service, believe me I think this usually doesn't make sense as well, but it does unfortunately happen.

I recall testing this a couple of times before as well, when I first found out about this. [It is actually an old issue which I always correct every time I make an update, it is just that I missed this one time to do it again](https://github.com/Scirra/Construct-3-bugs/issues/4565).

I don't know why it happens exactly, I do however know how the conflict happens. 
The probable cause is because the Cordova plugin author hardcoded the Google Services configuration in the `plugin.xml`, the plugin-side. Whereas Construct 3 configures in the `config.xml`, the Cordova project-side. I believe there is no right way to do this, it is more of a preference than a standard. However, both don't seem to mix well.

And, there is no real way to predict how Cordova plugins would do things, especially if they want to set the safe versions for their plugins.
I've tried keeping the tag I mentioned in the original post, and not setting the `GradlePluginGoogleServicesEnabled`. However, that still created a build error.

Hence, the only solution I saw working practically is to remove the tag I mentioned in the original post from the `plugin.xml` and let Construct 3 set the `GradlePluginGoogleServices`. Which works fine, I just forgot to reflect that in the latest update I have requested for whitelisting.

I hope that clears things up.

**Comment by Chadori** [comment]
Just in case it might help for future purposes, a Cordova plugin that has the mentioned tag in the original post, included in a Cordova plugin. 

The Cordova CLI builds successfully but the Construct 3 Build Service doesn't.

However though, I think regardless if it builds or not, having 2 Google Play Services declarations doesn't make sense. Therefore, I think it is justifiable for the Construct 3 Build Service to not complete the build, and the Cordova plugin needs to be adjusted regardless the case. Otherwise, the precedence of which declaration is used might not be clear.

**Cordova CLI**
![image](https://user-images.githubusercontent.com/31282960/159950789-0d3f9c9d-35fa-4b7f-85a5-6d5b7fbc1035.png)


**Construct 3 Build Service**
![image](https://user-images.githubusercontent.com/31282960/159951133-560a8e27-6ec0-4f3b-9e8f-66dae21308c9.png)

**Comment by Chadori** [comment]
Hi @AshleyScirra, that's great news then. That's one less reason to not make a fork.

The last one reason for having a fork is this: https://www.construct.net/en/forum/construct-3/plugin-sdk-10/request-specifying-resource-168426, ,as mentioned [here](https://github.com/Scirra/Construct-3-bugs/issues/5538).

**Comment by Chadori** [comment]
I really want to stop making forks as well, however there is still one more reason I had to fork, unfortunately. 


> Maybe you explained this before but I want to try to make sure forks aren't necessary

I recommended previously, if it is alright, maybe the editor can insert a `<resource-file>` in the `config.xml` of an exported Cordova project if files `google-services.json` and `GoogleService-Info.plist` are detected? To support most if not all Firebase Cordova plugins available.

This way we won't have to 1) Fork a Cordova plugin or 2) Create a new plugin for the sole purpose of running a hooks script to transfer the files `google-services.json` and `GoogleService-Info.plist` to their build locations.

However, I'm fine with option two (2) as well, I can make a small Cordova plugin if it is the most practical approach. I'm just curious if less plugins used is better since the common approach now is to include a `<resource-file>` in the config.xml of a Cordova project.
